src/util/newconfig/ConfigFileJson.cpp

```diff
@@ -0,0 +1,132 @@
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL,  DIRECT,  INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include "util/newconfig/ConfigFileJson.hpp"
+
+#include "util/Assert.hpp"
+#include "util/log/Logger.hpp"
+
+#include <boost/filesystem/path.hpp>
+#include <boost/json/array.hpp>
+#include <boost/json/object.hpp>
+#include <boost/json/parse.hpp>
+#include <boost/json/parse_options.hpp>
+#include <boost/json/value.hpp>
+
+#include <cstddef>
+#include <cstdint>
+#include <exception>
+#include <fstream>
+#include <ios>
+#include <iostream>
+#include <ostream>
+#include <sstream>
+#include <string>
+#include <string_view>
+#include <variant>
+#include <vector>
+
+namespace util::config {
+
+void
+ConfigFileJson::parse(boost::filesystem::path configFilePath)
+{
+    try {
+        std::ifstream const in(configFilePath.string(), std::ios::in | std::ios::binary);
+        if (in) {
+            std::stringstream contents;
+            contents << in.rdbuf();
+            auto opts = boost::json::parse_options{};
+            opts.allow_comments = true;
+            auto const tempObj = boost::json::parse(contents.str(), {}, opts).as_object();
+            flattenJson(tempObj, "");
+        }
+    } catch (std::exception const& e) {
+        LOG(util::LogService::error()) << "Could not read configuration file from '" << configFilePath.string()
+                                       << "': " << e.what();
+    }
+}
+
+std::variant<int64_t, std::string, bool, double>
+ConfigFileJson::getValue(std::string_view key) const
+{
+    ASSERT(jsonObject_.contains(key), "Json object does not contain key {}", key);
+    auto const jsonValue = jsonObject_.at(key);
+
+    auto const value = extractJsonValue(jsonValue);
+    return value;
+}
+
+std::vector<std::variant<int64_t, std::string, bool, double>>
+ConfigFileJson::getArray(std::string_view key) const
+{
+    ASSERT(jsonObject_.contains(key), "Key {} must exist in Json", key);
+    ASSERT(jsonObject_.at(key).is_array(), "Key {} has value that is not an array", key);
+
+    std::vector<std::variant<int64_t, std::string, bool, double>> configValues;
+    auto const arr = jsonObject_.at(key).as_array();
+
+    for (auto const& item : arr) {
+        auto const value = extractJsonValue(item);
+        configValues.emplace_back(value);
+    }
+    return configValues;
+}
+
+bool
+ConfigFileJson::containsKey(std::string_view key) const
+{
+    return jsonObject_.contains(key);
+}
+
+void
+ConfigFileJson::flattenJson(boost::json::object const& obj, std::string const& prefix)
```

Comment:
Does this method create additional json values inside parsed json with keys from config definition?

Reply:
It only creates key's of what the user has in their own local config. If some keys do not match clioConfig, parse will catch it and error out.